### PR TITLE
Use inventory and env private_data_dir subfolders

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
This is another part of trying to get AWX to follow `ansible-runner` best practices.

See docs:

https://ansible-runner.readthedocs.io/en/stable/intro.html#runner-input-directory-hierarchy

So what runner & people expect is:

```
.
├── env
│   ├── envvars
│   ├── extravars
│   ├── passwords
│   ├── cmdline
│   ├── settings
│   └── ssh_key
├── inventory
│   └── hosts
└── project
    ├── test.yml
    └── roles
        └── testrole
            ├── defaults
            ├── handlers
            ├── meta
            ├── README.md
            ├── tasks
            ├── tests
            └── vars
```

What we were producing was something more like

```
├── cp
├── env
│   ├── cmdline
│   ├── envvars
│   ├── extravars
│   ├── passwords
│   ├── settings
│   └── ssh_key
├── project
│   ├── ansible.cfg
│   ├── ansible_env.yml
│   ├── async_tasks.yml
<snip>
│   └── vault.yml
└── tmp3evnorsu
```

There was no inventory folder, and inventory was stored in `tmp3evnorsu` as a top-level file. Any credential files were also saved in that same structure.

With this change, it's more like:

```
├── cp
├── env
│   ├── cmdline
│   ├── envvars
│   ├── extravars
│   ├── passwords
│   └── settings
├── inventory
│   └── tmp94xjm0d7
└── project
    ├── ansible.cfg
    ├── ansible_env.yml
    ├── async_tasks.yml
<snip>
    └── vault.yml
```

and any credential files (starting with `tmp`) are dumped into the `env` folder.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION
```
14.1.0
```

##### ADDITIONAL INFORMATION
The motivation for this is that `ansible-runner` may require specific mounts for every file of directory in the private_data_dir. As such, we really don't want randomized filenames in the top level directory. I don't know for sure this is the direction we are going to go, but this seems like a defensible change on its own, to better adhere to the practices someone familiar with `ansible-runner` would expect.

### CODE REVIEW

**Comment by AlanCoding** [comment]
recheck

File: awx/main/tasks.py [hunk 1]
```diff
@@ -1478,10 +1479,6 @@ def run(self, pk, **kwargs):
                     module_args = ansible_runner.utils.args2cmdline(
                         params.get('module_args'),
                     )
-                shutil.move(
-                    params.pop('inventory'),
-                    os.path.join(private_data_dir, 'inventory')
-                )
```
Comment (AlanCoding) [inline awx/main/tasks.py]: I want this PR, if nothing else, for this. I hit a lot of errors related to container groups with execution environments that came back to this inventory move... which is something we shouldn't need and shouldn't exist.

**Review by chrismeyersfsu — APPROVED** [review]

**Comment by ryanpetrello** [comment]
`<low whistle>` this PR is pretty old.

@AlanCoding @shanemcd does this make sense as something to throw into the EE branch while we're here, or do you think it would require an excessive amount of rebasing at this point?

**Comment by AlanCoding** [comment]
I'm leaning towards letting `execution-environments` merge and then rebasing this. _This_ isn't that much content to rebase.

**Comment by ryanpetrello** [comment]
@AlanCoding (when you're back) this might be good to rebase and work on getting in.

**Comment by AlanCoding** [comment]
Hit friction in the rebase

```
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.8/site-packages/awx/main/tasks.py", line 1377, in run
    'inventory': self.build_inventory(self.instance, private_data_dir),
  File "/var/lib/awx/venv/awx/lib64/python3.8/site-packages/awx/main/tasks.py", line 1072, in build_inventory
    os.makedirs(path, mode=0o700)
  File "/usr/lib64/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/tmp/pdd_wrapper_19_w9mwdkjf/awx_19_0ja9wg3g/inventory'
```

That's due to deeper assumptions about directory creation order. I believe the `os.makedirs(path, mode=0o700)` line just gets deleted for this reason.

File: awx/main/models/credential/injectors.py [hunk 2]
```diff
@@ -96,25 +97,25 @@ def _openstack_data(cred):
...
     env['K8S_AUTH_API_KEY'] = cred.get_input('bearer_token', default='')
     if cred.get_input('verify_ssl') and 'ssl_ca_cert' in cred.inputs:
         env['K8S_AUTH_VERIFY_SSL'] = 'True'
-        handle, path = tempfile.mkstemp(dir=private_data_dir)
+        handle, path = tempfile.mkstemp(dir=os.path.join(private_data_dir, 'env'))
         with os.fdopen(handle, 'w') as f:
             os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
             f.write(cred.get_input('ssl_ca_cert'))
-        env['K8S_AUTH_SSL_CA_CERT'] = os.path.join('/runner', os.path.basename(path))
+        env['K8S_AUTH_SSL_CA_CERT'] = os.path.join('/runner', 'env', os.path.basename(path))
```
Comment (AlanCoding) [inline awx/main/models/credential/injectors.py]: I'm much less happy with these than I thought I would be.

There are tons of places in code where we put something in a directory, and other part of code relies on knowledge of how the directory & naming of that file was done. I would like something more robust to changes, and harder to get wrong for the host / container paths.

File: awx/main/tests/unit/test_tasks.py [hunk 3]
```diff
@@ -1053,7 +1068,7 @@ def test_gce_credentials(self, private_data_dir, job):
         safe_env = {}
         credential.credential_type.inject_credential(credential, env, safe_env, [], private_data_dir)
         runner_path = env['GCE_CREDENTIALS_FILE_PATH']
-        local_path = os.path.join(private_data_dir, os.path.basename(runner_path))
+        local_path = runner_path.replace('/runner', private_data_dir)  # container path to host path
```
Comment (AlanCoding) [inline awx/main/tests/unit/test_tasks.py]: I know these are only the tests, but this looks like the direction I would like to start taking things in.

There _might_ be corner cases that this will have problems with... but in the big picture, this is a durable way to switch from the container to host path.

File: awx/main/models/credential/injectors.py (hunk 2 above)
Comment (AlanCoding) [reply awx/main/models/credential/injectors.py]: So, to illustrate this point, let me write logic similar to what I've done in tests here

```python
env['K8S_AUTH_SSL_CA_CERT'] = path.replace(private_data_dir, '/runner')
```

Now, maybe this would be best implemented in a utility method (for better future control and error handling).

In general terms, this is where I'm interested in going, and I believe something like this would be hugely valuable for maintainability.

**Comment by AlanCoding** [comment]
I'm happy with how the code here looks now.
